lib/network_utils.ml

```diff
@@ -1179,8 +1179,16 @@ module Ovs = struct
         List.flatten (List.map (fun vif -> create_port_arg ?ty:(List.assoc_opt vif ifaces_with_type) vif name) existing_vifs)
       in
       let del_old_arg =
-        if vlan <> None then
-          (* This is to handle the case that a "real" bridge (not a "fake" VLAN bridge) already exists *)
+        let real_bridge_exists () =
+          try
+            (* `ovs-vsctl br-to-vlan <name>` returns 0 if <name> is a current "real" bridge *)
+            vsctl ~log:false ["br-to-vlan"; name] |> String.trim = "0"
```

Comment:
Manpage says that it will print `0` for a real bridge (note that this is not an exit-code, it is an output string),
but what about a VLAN with id of 0?

Reply:
Good spot. I've switched to `br-to-parent` now, which avoids the problem.